src/Renderer/LayeredMaterial.js

```diff
@@ -177,152 +187,248 @@ LayeredMaterial.prototype.dispose = function() {
+};
+
+LayeredMaterial.prototype.getTextureCountByLayerIndex = function(index) {
+    return this.layerTexturesCount[index];
+};
+
+LayeredMaterial.prototype.getLayerTextureOffset = function(layer) {
+    var index = this.indexOfColorLayer(layer);
+    return index > -1 ? this.getTextureOffsetByLayerIndex(index) : -1;
 };
 
 LayeredMaterial.prototype.setLightingOn = function(enable) {
-    this.uniforms.lightingOn.value = enable === true ? 1 : 0;
+    this.uniforms.lightingOn.value = enable;
 };
 
-LayeredMaterial.prototype.setLayerOpacity = function(id, opacity) {
+LayeredMaterial.prototype.setLayerFx = function(index, fx) {
+    this.uniforms.paramLayers.value[index].z = fx;
+};
 
-    if (this.paramLayers[id]) {
-        this.paramLayers[id].w = opacity;
-    }
+LayeredMaterial.prototype.setLayerOffset = function(index, offset) {
+    this.uniforms.paramLayers.value[index].x = offset;
 };
 
-LayeredMaterial.prototype.setLayerVisibility = function(id, visible) {
+LayeredMaterial.prototype.setLayerUV = function(index, idUV) {
+    this.uniforms.paramLayers.value[index].y = idUV;
+};
 
-    if (this.paramLayers[id]) {
-        this.paramLayers[id].z = visible ? 1 : 0;
-    }
+LayeredMaterial.prototype.setLayerOpacity = function(index, opacity) {
+    if(this.uniforms.paramLayers.value[index])
+        this.uniforms.paramLayers.value[index].w = opacity;
+};
+
+LayeredMaterial.prototype.setLayerVisibility = function(index, visible) {
+    this.uniforms.visibility.value[index] = visible;
+};
+
+LayeredMaterial.prototype.setLayerTexturesCount = function(index, count) {
+    this.layerTexturesCount[index] = count;
+};
+
+LayeredMaterial.prototype.loadedTexturesCount = function() {
+    return this.texturesCount[l_ELEVATION] + this.texturesCount[l_COLOR];
 };
 
-LayeredMaterial.prototype.setNbLayersColor = function(n) {
-    this.uniforms.nColorLayer.value = n;
+LayeredMaterial.prototype.downScaledColorLayer = function(layer,level) {
+    return this.Textures[l_COLOR][this.getLayerTextureOffset(layer)].level < level;
+};
+
+LayeredMaterial.prototype.downScaledLayer = function(layerType,level) {
+
+    if (layerType === l_ELEVATION) {
+        if (this.Textures[l_ELEVATION][0].level < 0) {
+            return false;
+        } else {
+            return this.Textures[l_ELEVATION][0].level < level;
+        }
+    } else if (layerType === l_COLOR) {
+        for (var index = 0, max = this.colorLayersId.length; index < max; index++) {
+            let offset = this.getTextureOffsetByLayerIndex(index);
+            if (this.Textures[l_COLOR][offset].level < level) {
+                return true;
+            }
+        }
+    }
+
+    return false;
 };
 
+LayeredMaterial.prototype.getColorLayerLevel = function(colorLayerId) {
 
-LayeredMaterial.prototype.getLayerLevel = function(id, slot) {
-    var level = this.Textures[id][slot || 0].level;
+    let index = this.indexOfColorLayer(colorLayerId);
+    if(index ===-1) {
```

Comment:
I'm not sure that hiding the "colorLayerId is invalid" error is a good idea.
Either throw an error, or return an invalid level value (`-1` ?)